x-pack/plugins/fleet/server/routes/package_policy/handlers.ts

```diff
@@ -425,6 +426,8 @@ export const updatePackagePolicyHandler: FleetRequestHandler<
       throw new PackagePolicyRequestError('At least one agent policy id must be provided');
     }
 
+    await renameAgentlessAgentPolicy(soClient, esClient, packagePolicy, newData.name);
```

Comment:
Some tests would be good as it is on a critical path of updating policy. Unit for the util itself and unit/integration for `updatePackagePolicyHandler`

Reply:
tests added, Good call!